demo.md

```diff
@@ -0,0 +1,155 @@
+`Ensure that you have finished setup before doing this!`
+
+1. **Identify** **topics** and **frame_id** which your **Depth Sensor/Camera** is publishing the following information on:
+
+> `3D PointCloud`
+>
+> `Rectified 2D RGB Image`
+
+Eg. In my own use case, I use a Kinect v2 One Depth Sensor. The **topics** and **frame_id** thus look like so:
+> `/kinect2/hd/points` - For 3D PointCloud
+
+> `/kinect2/hd/image_color_rect` - For Rectified 2D RGB Image
+
+> `/kinect2_rgb_optical_frame` - For frame_id
+
+2 . **Replace** the following lines in RVIP's launch file under the specified files below:
+
+**Edit** [line 20]() of the `rvip/launch/run.launch` file to the **3D PointCloud** camera topic you identified earlier.
+```xml
+<remap from="camera_3d" to="/kinect2/hd/points"/>
+```
+
+**Edit** [line 23]() of the `rvip/launch/run.launch` file to the **frame_id** you identified earlier.
+```xml
+<param name="frame_id" value="/kinect2_rgb_optical_frame" />
+```
+
+**Edit** [line 4]() of the `darknet_ros/darknet_ros/config/ros.yaml` file to the **Rectified 2D RGB Image** you identified earlier.
+```yaml
+    topic: /kinect2/hd/image_color_rect
+```
+
+3. **Run** the command below to launch **RVIP**.
+> `$ roslaunch rvip run.launch`
+
+# **Setup**
+
+### There are **3** parts to the setup.
+1. OpenGR Setup
+2. darknet_ros Setup
+3. RVIP Setup
+
+# !**`WARNING`**!
+> [WIP] The **OpenGR** library used in 0.0.1 alpha version is an outdated version. Please follow the instructions below to download the specific version of OpenGR highlighted.
+
+### **[OpenGR Setup]**
+
+1. **Download** the OpenGR library source codes from its GitHub repo.
+> `$ git clone https://github.com/STORM-IRIT/OpenGR.git`
+
+2. **Go** into the **OpenGR** directory
+> `$ cd OpenGR`
+3. **Git Revert** to the specified version of OpenGR.
+> `$ git reset --hard 0967cd880950b35786b8fd098837c9eb1fe2aca4`
+
+![](img/opengr_commit.png)
+**Click** [here](https://github.com/STORM-IRIT/OpenGR/commit/0967cd880950b35786b8fd098837c9eb1fe2aca4) to go to the webpage with this commit.
+
+4. **Create** and **Go** to a build directory
+> `$ mkdir build && cd build`
+
+5. **Build** the **OpenGR** using cmake.
+> `$ cmake -DCMAKE_BUILD_TYPE=Release ..`
+
+6. **Install** the **OpenGR** into your local workstation.
+> `$ sudo make install`
+
+### **[darknet_ros Setup]**
+
+1. **Go** to the `src` directory of your **catkin** workspace.
+> `$ cd path/to/your/<workspace_name>/src`
+
+2. **Download** the git repository into your local workstation.
+> `$ git clone --recursive https://github.com/leggedrobotics/darknet_ros.git`
+
+3. **Replace** the `CMakeLists.txt` under `darknet_ros/darknet_ros` with the  
+`CMakeLists.txt` from [this link](https://github.com/benjaminabruzzo/darknet_ros/blob/master/darknet_ros/CMakeLists.txt).
+
+This removes reliance of CUDA temporarily to run **darknet_ros** YOLO object detection.
```

Comment:
Is this desirable? Doesn't CUDA provide significant speedups?

Reply:
Adding CUDA would provide significant speedup **but** also provide further complications in setup.

Every laptop comes with a CPU, but not every laptop has a CUDA-enabled NVIDIA GPU.

Reply:
Perhaps add a short comment stating this? As a more experienced user, this line in `demo.md` confused me.

Reply:
Addressed in [this commit](https://github.com/ros-industrial/rvip/pull/1/commits/ae3d2f7b33bcd62fcee9e765a9a0982729432240).